client/components/drag_drop/index.coffee

```diff
@@ -74,7 +74,7 @@ module.exports = React.createClass
 
           DragTarget {
             key: i + '-' + child.type.displayName + '-target'
-            i: i
+            i: child.props.index or i
             setDragTarget: @setDragTarget
```

Comment:
Sometimes drag/drop accepts an array of object of mixed types -- i.e., section list included non-draggable section tools in the same array.  This makes sure that it uses the provided index if given, and ignore non-draggable items when returning the new order.